plugins/GCodeWriter/GCodeWriter.py

```diff
@@ -69,8 +70,13 @@ def _serialiseSettings(self, settings):
         prefix_length = len(prefix)
 
         global_stack = Application.getInstance().getGlobalContainerStack()
-        container_with_profile = global_stack.findContainer({"type": "quality"})
-        serialized = container_with_profile.serialize()
+        # Collecting all data from the top of the stack
+        temporarySettings= InstanceContainer("")
+        temporarySettings.setDefinition(ContainerRegistry.getInstance().findDefinitionContainers(id="fdmprinter")[0])
+        for key in global_stack.getAllKeys():
```

Comment:
`global_stack`.getAllKeys() gets all keys in FDMPrinter. You'd have to call this function on an extruder stack to get the keys in FDMExtruder. I believe those stacks are held in Cura/Settings/ExtruderManager.py.